### PR TITLE
habitat 0.7.0 (new formula)

### PR DESCRIPTION
**Pull request by bltavares**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Adds formula for [Habitat](https://www.habitat.sh/)

### CODE REVIEW

File: Formula/habitat.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Habitat < Formula
+  desc "Application automation framework to build applications"
+  homepage "https://www.habitat.sh/"
+
+  url "https://api.bintray.com/content/habitat/stable/darwin/x86_64/hab-0.6.0-20160613151520-x86_64-darwin.zip?bt_package=hab-x86_64-darwin"
```
Comment (schisamo) [inline Formula/habitat.rb]: You'll want to go directly to the CDN link of:
`https://habitat.bintray.com/stable/darwin/x86_64/hab-<VERSION>-x86_64-darwin.zip`

The Bintray API link is only useful if you want to use a logical version of `$latest` in your URL like we do on the main [Habitat download page](https://www.habitat.sh/docs/get-habitat/): `https://api.bintray.com/content/habitat/stable/darwin/x86_64/hab-$latest-x86_64-darwin.zip?bt_package=hab-x86_64-darwin` (note the `$latest`).

You'll also want to use the latest 0.7.0 release in this initial formula:

URL -> `https://habitat.bintray.com/stable/darwin/x86_64/hab-0.7.0-20160614231131-x86_64-darwin.zip`
SHA256 -> `93fac880261df34c75ac5775c9febb536b82ac955cd6df873bbfec1cf697ed50`

**Comment by schisamo** [comment]
As a side note for future updating you can get the SHA256 checksum by making a simple `HEAD` request to the download URL (this works for all Bintray URLs):

```
➜  curl -I https://habitat.bintray.com/stable/darwin/x86_64/hab-0.7.0-20160614231131-x86_64-darwin.zip
HTTP/1.1 200 OK
Server: nginx
Date: Wed, 15 Jun 2016 12:38:36 GMT
Content-Type: application/zip
Content-Length: 3149461
Connection: keep-alive
Last-Modified: Tue, 14 Jun 2016 23:57:08 GMT
Accept-Ranges: none
ETag: 93fac880261df34c75ac5775c9febb536b82ac955cd6df873bbfec1cf697ed50
X-Checksum-Sha1: 02ffd11436f26f84d580a5b419b68337a6230f2d
X-Checksum-Sha2: 93fac880261df34c75ac5775c9febb536b82ac955cd6df873bbfec1cf697ed50
```

File: Formula/habitat.rb (hunk 1 above)
Comment (bltavares) [inline Formula/habitat.rb]: Thanks for the information, I had trouble using the `$latest` syntax and just replaced with the name of the file as I didn't know the alternate URL

**Comment by stevenhaddox** [comment]
Just tried to install this on Linux via linuxbrew before reviewing the code. Apparently it's only for the OS X version of habitat. My bad. Super excited for my Macs at home though!

**Comment by DomT4** [comment]
This was added as a Cask, since it apparently doesn't build easily from source. Ref: https://github.com/caskroom/homebrew-cask/pull/21977

`brew cask install hab`.

**Pull request closed without merging**